### PR TITLE
Development and Deployment

### PR DESCRIPTION
**Pull request by TomDijkema**
Add:
- API functionality
- Optional taxa for /predict endpoint
- Predict table
- Predict legend
- Filter functionality for predict table based on prediction score
- Taxon details section
- Draft for interlinked taxa graph

Graph still needs work, commented code in that file can be ignored

### CODE REVIEW

File: src/api/predict/Predict.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import axios from 'axios';
+
+
+function Predict(formData, callback) {
+    console.log(formData);
```
Comment (samleeflang) [inline src/api/predict/Predict.js]: Remove console.log

File: src/templates/home/Test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import ForceGraph2D from 'react-force-graph-2d';
```
Comment (samleeflang) [inline src/templates/home/Test.js]: A bit strange name, Test.js what does this file do?

File: src/templates/results/components/GraphLayout.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                taxonA: node['id']
+            }, Process);
+
+            function Process(result) {
+                console.log(result);
+            }
+        }
+    }
+
+    // const graphData = {
```
Comment (samleeflang) [inline src/templates/results/components/GraphLayout.js]: Remove commented code

File: src/templates/results/components/GraphLayout.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    /* When clicked on node in grpah, try to extend graph with new found relations */
+    const ExtendUponNode = (node) => {
+        if (!keyNodeTracker.includes(node['id'])) {
+            PredictInteraction({
+                interaction: formData['interaction'],
+                taxonA: node['id']
+            }, Process);
+
+            function Process(result) {
+                console.log(result);
```
Comment (samleeflang) [inline src/templates/results/components/GraphLayout.js]: Check if console.log is needed

File: src/templates/results/components/ObservedTable.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        <Row className="h-100">
+            <Col className="h-100">
+                <Row className="results_titleBar">
+                    <Col className="results_titleBarSub col-md-auto fw-bold h-100">
+                        Observed interactions
+                    </Col>
+                </Row>
+                <Row className="results_tableSection">
+                    <Col className="h-100">
+                        <Table striped className="results_table h-100">
```
Comment (samleeflang) [inline src/templates/results/components/ObservedTable.js:18]: Probably not necessary here but there are a couple of good libs out there for tables. Worked with agGrid previously but another option could be the ant table checklist bank uses. Might be a bit overkill for here but know they are there.

File: src/templates/results/components/PredictedLegend.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import { Row, Col } from 'react-bootstrap';
+
+
+const PredictedLegend = (props) => {
```
Comment (samleeflang) [inline src/templates/results/components/PredictedLegend.js:4]: Seems like something that should be generated by the library. Why is it necessary to build the legend yourself?

Comment (samleeflang) [reply src/templates/results/components/PredictedLegend.js:4]: Ok next files explains it, might be nicer if we could get something out of the box for this but not necessary, and not for this PR.

**Review by samleeflang — COMMENTED** [review]
Couple of minor things

File: src/api/predict/Predict.js (hunk 1 above)
Comment (TomDijkema) [reply src/api/predict/Predict.js]: Removed

File: src/templates/home/Test.js (hunk 2 above)
Comment (TomDijkema) [reply src/templates/home/Test.js]: Removed file, was for testing the graph library, now moved to: GraphLayout.js

File: src/templates/results/components/GraphLayout.js (hunk 3 above)
Comment (TomDijkema) [reply src/templates/results/components/GraphLayout.js]: Removed

File: src/templates/results/components/GraphLayout.js (hunk 4 above)
Comment (TomDijkema) [reply src/templates/results/components/GraphLayout.js]: Removed

File: src/templates/results/components/ObservedTable.js (hunk 5 above)
Comment (TomDijkema) [reply src/templates/results/components/ObservedTable.js:18]: Good to know! Until now been using the Bootstrap Table, which is pretty simple. Could be more interesting for UCAS as it uses a library for table sorting which is outdated (and causes some issues with legacy peer deps)

File: src/templates/results/components/PredictedLegend.js (hunk 6 above)
Comment (TomDijkema) [reply src/templates/results/components/PredictedLegend.js:4]: Was thinking about it at first, could probably do it, also like to practice with these kind of things. Could do it if there is time in the future

Comment (samleeflang) [reply src/templates/results/components/PredictedLegend.js:4]: :+1:

**Review by samleeflang — APPROVED** [review]
All comments resolved

**Pull request merged**